### PR TITLE
Expand the transaction callback ordering in guides [ci-skip]

### PR DESCRIPTION
**Pull request by p8**
The ordering can now be configured.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/active_record_callbacks.md [hunk 1]
```diff
@@ -621,6 +623,15 @@ end
...
 NOTE: This applies to all `after_*_commit` variations too, such as `after_destroy_commit`.
 
+This order can be set via configuration:
+
+```ruby
+config.active_record.run_after_transaction_callbacks_in_order_defined = false
+```
+
+If `true` (the default from Rails 7.1), callbacks are executed in the order they
+are defined. If `false`, the order is reversed, just like in the example above.
```
Comment (akhilgkrishnan) [inline guides/source/active_record_callbacks.md:633]:
```suggestion
When set to `true` (the default from Rails 7.1), callbacks are executed in the order they
are defined. When set to `false`, the order is reversed, just like in the example above.
```

**Review by akhilgkrishnan — COMMENTED** [review]
Added a minor suggestion.

File: guides/source/active_record_callbacks.md (hunk 1 above)
Comment (p8) [reply guides/source/active_record_callbacks.md]: Thanks! That reads better indeed.
I'll also change it in the API docs where I copied it from.

**Review by akhilgkrishnan — APPROVED** [review]
Thanks @p8 for the changes 👍

**Review by akhilgkrishnan — APPROVED** [review]

**Comment by p8** [comment]
Thanks @akhilgkrishnan !

**Pull request merged**